packages/sanity/src/core/studio/components/navbar/search/components/filters/filter/inputs/date/datePicker/calendar/CalendarDay.tsx

```diff
@@ -1,51 +1,118 @@
-import {Card, Text} from '@sanity/ui'
+import {Card, Text, Theme} from '@sanity/ui'
+import {isAfter, isBefore, isSameDay, isSameMonth} from 'date-fns'
 import React, {useCallback} from 'react'
-import {useDatePicker} from '../contexts/useDatePicker'
+import styled, {css} from 'styled-components'
+import {useCalendar} from './contexts/useDatePicker'
 
 interface CalendarDayProps {
   date: Date
-  focused?: boolean
   onSelect: (date: Date) => void
-  isCurrentMonth?: boolean
-  isToday: boolean
-  selected?: boolean
 }
 
-export function CalendarDay(props: CalendarDayProps) {
-  const {date, focused, isCurrentMonth, isToday, onSelect, selected} = props
+const CircleSvg = styled.svg(({theme}: {theme: Theme}) => {
+  return css`
+    bottom: 0;
+    left: 0;
+    position: absolute;
+    right: 0;
+    top: 0;
 
+    circle {
+      stroke: ${theme.sanity.color.card.enabled.border};
+      stroke-width: 3;
+      fill: none;
+    }
+  `
+})
+
+const CustomCard = styled(Card)<{
+  $isEndDate?: boolean
+  $isStartDate?: boolean
+  $isWithinRange?: boolean
+}>(({$isEndDate, $isStartDate, $isWithinRange, theme}) => {
+  const borderRadius = $isWithinRange
+    ? {left: 0, right: 0}
+    : {
+        left: $isEndDate && !$isStartDate ? 0 : `${theme.sanity.radius[2]}px`,
+        right: $isStartDate && !$isEndDate ? 0 : `${theme.sanity.radius[2]}px`,
+      }
+  return css`
+    &[data-focused='true'] {
+      z-index: 1;
+    }
+    border-radius: ${borderRadius.left} ${borderRadius.right} ${borderRadius.right}
+      ${borderRadius.left};
+    position: relative;
+  `
+})
+
+export function CalendarDay({date, onSelect}: CalendarDayProps) {
   const handleClick = useCallback(() => {
     onSelect(date)
   }, [date, onSelect])
 
-  const {fontSize} = useDatePicker()
+  const {
+    date: selectedDate,
+    endDate: selectedEndDate,
+    focusedDate,
+    fontSize,
+    selectRange,
+  } = useCalendar()
+
+  const isSelected = selectedDate && isSameDay(date, selectedDate)
+  const isStartDate = selectRange && selectedDate && isSameDay(date, selectedDate)
+  const isEndDate = selectRange && selectedEndDate && isSameDay(date, selectedEndDate)
+
+  const isCurrentMonth = isSameMonth(date, focusedDate)
+  const isFocused = focusedDate && isSameDay(date, focusedDate)
+  const isToday = isSameDay(date, new Date())
+
+  const isWithinRange =
+    selectedDate &&
+    selectedEndDate &&
+    !isStartDate &&
+    !isEndDate &&
+    isAfter(date, selectedDate) &&
+    isBefore(date, selectedEndDate)
 
   return (
-    <div aria-selected={selected} data-ui="CalendarDay">
-      <Card
-        aria-label={date.toDateString()}
-        aria-pressed={selected}
-        as="button"
+    <div aria-selected={isSelected} data-ui="CalendarDay">
```

Comment:
Is this div needed, or could it be removed and instead add the aria and data attribute to the `CustomCard`?